src/sql/workbench/parts/notebook/common/models/notebookInput.ts

```diff
@@ -41,9 +46,17 @@ export class NotebookEditorModel extends EditorModel {
 		this._register(this.notebookService.onNotebookEditorAdd(notebook => {
 			if (notebook.id === this.notebookUri.toString()) {
 				// Hook to content change events
-				notebook.modelReady.then(() => {
-					this._register(notebook.model.kernelChanged(e => this.updateModel()));
-					this._register(notebook.model.contentChanged(e => this.updateModel(e)));
+				notebook.modelReady.then((model) => {
+					if (!this.changeEventsHookedUp) {
+						this.changeEventsHookedUp = true;
```

Comment:
Why would this be fired multiple times?

Reply:
This is a bug that I need to investigate where the modelReady event can be triggered multiple times when the same notebook is closed/opened multiple times. This isn't a regression (and this condition improves things), so I'd rather fix that issue in another PR. Filed an issue to track it here: https://github.com/microsoft/azuredatastudio/issues/6884